src/adapters/store/postgres.ts

```diff
@@ -109,50 +103,54 @@ export function createPgStoreAdapter(
   }
 
   async function lastEventOf<E extends Event<any, any>>(
-    eventType: EventNamespaceAndType,
+    ns: EventNamespace,
+    ty: EventType
   ): Promise<Option<E>> {
-    logger.trace({ eventType }, 'eventStoreLastEventOfResult');
+    logger.trace({ ns, ty }, "eventStoreLastEventOf");
 
     const start = Date.now();
 
     return pool
       .query(
-        `select * from events where data->>'type' = $1 order by context->>'time' desc limit 1`,
-        [eventType],
+        `select * from events
+        where data->>'event_namespace' = $1
+        and data->>'event_type' = $2
+        order by context->>'time' desc limit 1`,
+        [ns, ty]
```

Comment:
Do we need to care about indices? I understand this is not something to solve in this PR, but should we think about it? Or maybe document it in the StoreAdapter description so the implementer knows about potential DBA optimizations.

Reply:
We should definitely explore what the right index would be. I won't touch anything in this PR - we should do some benchmarking before adding docs or more functionality.